R/derive_param_tte.R

```diff
@@ -142,10 +159,259 @@
+#'   "01",     ymd("2021-03-04"),      2, "Cough",
+#'   "01",     ymd("2021-03-05"),      3, "Cough"
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
+#'
+#' ttae <- event_source(
+#'   dataset_name = "ae",
+#'   date = AESTDT,
+#'   set_values_to = exprs(
+#'     EVNTDESC = "AE",
+#'     SRCDOM = "AE",
+#'     SRCVAR = "AESTDTC",
+#'     SRCSEQ = AESEQ
+#'   )
+#' )
+#'
+#' eos <- censor_source(
+#'   dataset_name = "adsl",
+#'   date = EOSDT,
+#'   set_values_to = exprs(
+#'     EVNTDESC = "END OF STUDY",
+#'     SRCDOM = "ADSL",
+#'     SRCVAR = "EOSDT"
+#'   )
+#' )
+#'
+#' derive_param_tte(
+#'   dataset_adsl = adsl,
+#'   event_conditions = list(ttae),
+#'   censor_conditions = list(eos),
+#'   source_datasets = list(adsl = adsl, ae = ae),
+#'   set_values_to = exprs(
+#'     PARAMCD = "TTAE",
+#'     PARAM = "Time to First Adverse Event"
+#'   )
+#' ) %>%
+#'   select(USUBJID, STARTDT, PARAMCD, PARAM, ADT, CNSR, SRCSEQ)
+#'
+#' @caption Adding a by variable (`by_vars`)
+#' @info By variables can be added using the `by_vars` argument, e.g., now for
+#'   each subject the time to first occurrence of each adverse event preferred
+#'   term (`AEDECOD`) should be created as parameters.
+#' @code
+#' derive_param_tte(
+#'   dataset_adsl = adsl,
+#'   by_vars = exprs(AEDECOD),
+#'   event_conditions = list(ttae),
+#'   censor_conditions = list(eos),
+#'   source_datasets = list(adsl = adsl, ae = ae),
+#'   set_values_to = exprs(
+#'     PARAMCD = paste0("TTAE", as.numeric(as.factor(AEDECOD))),
+#'     PARAM = paste("Time to First", AEDECOD, "Adverse Event")
+#'   )
+#' ) %>%
+#'   select(USUBJID, STARTDT, PARAMCD, PARAM, ADT, CNSR, SRCSEQ)
+#'
+#' @caption Handling duplicates (`check_type`)
+#' @info The source records are checked regarding duplicates with respect to the
+#'   by variables and the date and order specified in the source objects.
+#'   By default, a warning is issued if any duplicates are found.
+#'   Note here how the `source_datasets` argument was updated to read in the
+#'   new adverse event source dataset constructed below containing a duplicate
+#'   date for `"Cough"`.
+#' @code [expected_cnds = "duplicate_records"]
+#' ae_dup <- tribble(
+#'   ~USUBJID, ~AESTDT,           ~AESEQ, ~AEDECOD, ~AESER,
+#'   "01",     ymd("2021-01-03"),      1, "Flu",    "Y",
+#'   "01",     ymd("2021-03-04"),      2, "Cough",  "N",
+#'   "01",     ymd("2021-03-04"),      3, "Cough",  "Y"
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
+#'
+#' derive_param_tte(
+#'   dataset_adsl = adsl,
+#'   by_vars = exprs(AEDECOD),
+#'   start_date = TRTSDT,
+#'   source_datasets = list(adsl = adsl, ae = ae_dup),
+#'   event_conditions = list(ttae),
+#'   censor_conditions = list(eos),
+#'   set_values_to = exprs(
+#'     PARAMCD = paste0("TTAE", as.numeric(as.factor(AEDECOD))),
+#'     PARAM = paste("Time to First", AEDECOD, "Adverse Event")
+#'   )
+#' )
+#'
+#' @info For investigating the issue the dataset of the duplicate source records
```

Comment:
```suggestion
#' @info For investigating the issue the dataset of the duplicate source records
```

ehhh sorry this is hard to follow.  can it be re-worded??